bin/monitor.ml

```diff
@@ -219,21 +219,59 @@ let render_loop ~(event : Event.t Fiber_event_bus.t) =
   loop ()
 ;;
 
-let monitor () =
-  let where = Rpc_common.active_server () in
+let run_fetchers ~event ~client =
+  let module Sub = Dune_rpc_private.Public.Sub in
+  Fiber.all_concurrently_unit
+    [ fetch_loop ~event ~client ~f:(fun x -> Event.Jobs x) Sub.running_jobs
+    ; fetch_loop ~event ~client ~f:(fun x -> Event.Progress x) Sub.progress
+    ; fetch_loop ~event ~client ~f:(fun x -> Event.Diagnostics x) Sub.diagnostic
+    ]
+;;
+
+let repeat_while_counting ~init ~f =
```

Comment:
This can be implemented on top of `Fiber.repeat_while`.

Reply:
The issue is that `Fiber.repeat_while` returns `unit` when given `None`, but I want to get the value of `Where.get ()`. I could use it, but then I would have to repeat `Where.get`.

Reply:
You would only return `None` when `auto_poll = false`, right? So I don't see why `Where.get` would be repeated.

Reply:
Oh I see how to use it now, will do. I wasn't thinking about the whole function but just the where call.